src/libYARP_dev/src/devices/RobotDescriptionServer/RobotDescriptionServer.cpp

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright (C) 2016 iCub Facility - Istituto Italiano di Tecnologia
+ * Author: Marco Randazzo <marco.randazzo@iit.it>
+ * CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ */
+
+#include <yarp/dev/IRobotDescription.h>
+#include "RobotDescriptionServer.h"
+#include <yarp/os/Log.h>
+#include <yarp/os/LogStream.h>
+#include <yarp/os/LockGuard.h>
+
+/*! \file Navigation2DClient.cpp */
+
+using namespace yarp::dev;
+using namespace yarp::os;
+using namespace yarp::sig;
+
+
+//------------------------------------------------------------------------------------------------------------------------------
+
+bool yarp::dev::RobotDescriptionServer::open(yarp::os::Searchable &config)
+{
+    m_local_name.clear();
+    m_local_name = config.find("local").asString().c_str();
+
+    if (m_local_name == "")
+    {
+        yError("RobotDescriptionServer::open() error you have to provide valid local name");
+        return false;
+    }
+
+    ConstString local_rpc = m_local_name;
+    local_rpc += "/rpc";
+
+    if (!m_rpc_port.open(local_rpc.c_str()))
+    {
+        yError("RobotDescriptionServer::open() error could not open rpc port %s, check network", local_rpc.c_str());
+        return false;
+    }
+
+    m_rpc_port.setReader(*this);
+    return true;
+}
+
+bool yarp::dev::RobotDescriptionServer::attachAll(const PolyDriverList &p)
+{
+    for (int i = 0; i < p.size(); i++)
+    {
+        //yDebug() << "***************" << p[i]->poly->getOptions().toString();
+        //yDebug() << "***************" << p[i]->poly->getValue("device").toString();
+        //yDebug() << "***************" << p[i]->poly->getValue("name").toString();
+        RobotDescription dev;
+        dev.device_name = p[i]->poly->getValue("name").toString();
```

Comment:
After discussion with  @barbalberto  and @traversaro we decided to merge in devel and test the current implementation. In January, we'll refine it, for example adding suitable methods (e.g. get_type() get_name()) to yarp::dev::DeviceDriver class (this will require modifications to all yarp device drivers.).